apisix/core/utils.lua

```diff
@@ -292,11 +292,16 @@ local resolve_var
 do
     local _ctx
     local n_resolved
-    local pat = [[(?<!\\)\$\{?(\w+)\}?]]
+    local pat = [[(?<!\\)\$(\{(\w+)\}|(\w+))]]
     local _escaper
 
     local function resolve(m)
-        local v = _ctx[m[1]]
+        local i = 1
+        -- if first capture's first char is "{" evaluate the second capture
+        if m[1]:byte(1) == 123 then
+            i = 2
+        end
+        local v = _ctx[m[i]]
```

Comment:
Why do you add this?

Reply:
this was to fix other test case failures. For example when var = `${alpha}`, m[1] = {alpha} and m[2] = alpha. Only m[2] can be evaluated no m[1].